lib/ts/recipe/webauthn/types.ts

```diff
@@ -0,0 +1,858 @@
+    // to be called just during sign up. But we also need a version of signing up which can be
+    // called during operations like creating a user during password reset flow.
+    createNewRecipeUser(input: {
+        webauthnGeneratedOptionsId: string;
+        credential: {
+            id: string;
+            rawId: string;
+            response: {
+                clientDataJSON: string;
+                attestationObject: string;
+                transports?: ("ble" | "cable" | "hybrid" | "internal" | "nfc" | "smart-card" | "usb")[];
+                userHandle: string;
+            };
+            authenticatorAttachment: "platform" | "cross-platform";
+            clientExtensionResults: Record<string, unknown>;
+            type: "public-key";
+        };
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<
+        | {
+              status: "OK";
+              user: User;
+              recipeUserId: RecipeUserId;
+          }
+        | CreateNewRecipeUserErrorResponse
+    >;
+
+    verifyCredentials(input: {
+        webauthnGeneratedOptionsId: string;
+        credential: {
+            id: string;
+            rawId: string;
+            response: {
+                clientDataJSON: string;
+                attestationObject: string;
+                transports?: ("ble" | "cable" | "hybrid" | "internal" | "nfc" | "smart-card" | "usb")[];
+                userHandle: string;
+            };
+            authenticatorAttachment: "platform" | "cross-platform";
+            clientExtensionResults: Record<string, unknown>;
+            type: "public-key";
+        };
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<{ status: "OK"; user: User; recipeUserId: RecipeUserId } | VerifyCredentialsErrorResponse>;
+
+    // used for retrieving the user details (email) from the recover account token
+    // should be used in the registerOptions function when the user recovers the account and generates the credentials
+    getUserFromRecoverAccountToken(input: {
+        token: string;
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<{ status: "OK"; user: User; recipeUserId: RecipeUserId } | GetUserFromRecoverAccountTokenErrorResponse>;
+
+    // credentials CRUD
+    removeCredential(input: {
+        webauthnCredentialId: string;
+        recipeUserId: RecipeUserId;
+        userContext: UserContext;
+    }): Promise<
+        | {
+              status: "OK";
+          }
+        | RemoveCredentialErrorResponse
+    >;
+
+    getCredential(input: {
+        webauthnCredentialId: string;
+        recipeUserId: RecipeUserId;
+        userContext: UserContext;
+    }): Promise<
+        | {
+              status: "OK";
+              credential: {
+                  id: string;
+                  rp_id: string;
+                  created_at: number;
+              };
+          }
+        | RemoveCredentialErrorResponse
```

Comment:
why does the get function return a RemoveCredentialErrorResponse?

Reply:
Also, this function should return the recipeUserId that uses this credential as well.

Reply:
done